Dockerfile

```diff
@@ -7,9 +7,16 @@ ENV PATH /go/bin:/usr/local/go/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr
 ARG COMMIT_SHA
 ARG ENVIRONMENT
 
-ADD . /go/src/github.com/smartcontractkit/chainlink
+# Do dependency installs first, since these will change less than the full
+# source tree and can get cached
 WORKDIR /go/src/github.com/smartcontractkit/chainlink
-RUN make build
+ADD Gopkg.* /go/src/github.com/smartcontractkit/chainlink/
+RUN dep ensure -vendor-only
```

Comment:
Before these changes `make build` ran the following commands:
```
godep
yarndep
...
```

And now with these new steps, we manually run the same steps but in Docker. Is it to leverage docker's caching mechanism so subsequent builds don't re-execute `dep ensure`?

If so, let's use the `make` commands instead of manually invoking `dep ensure -vendor-only`:

```
RUN make godep
...
RUN make yarndep
```

Reply:
Yes, good idea!